packages/dds/fix_wn_bin_2.2.patch

```diff
@@ -0,0 +1,185 @@
+-
+-#
+-# TODO: replace DDS_CollectPrerequisites.cmake by DDS_CollectPrerequisitesGen.cmake and make the script more generic
+-#
+ add_custom_target( wn_bin
+-	COMMAND -mkdir -p ${WN_PKG_DIR}
+-	COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:dds-agent> "${WN_PKG_DIR}"
+-	COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:dds-user-defaults> "${WN_PKG_DIR}"
+-	COMMAND ${CMAKE_COMMAND} -DPREREQ_DESTINATION=${WN_PKG_DIR} -DDDS_AGENT_BIN_PATH=${DDS_AGENT_BIN_PATH}
+-		-DPREREQ_DIRS=${PREREQ_DIRS} -DDDS_BOOST_LIB_DIR=${DDS_BOOST_LIB_DIR} -P "${CMAKE_SOURCE_DIR}/cmake/modules/DDS_CollectPrerequisites.cmake"
++	COMMAND ${CMAKE_COMMAND} -E make_directory ${WN_PKG_DIR}
++  COMMAND ${CMAKE_COMMAND}
++    -DEXECUTABLES=${CLI_WN_PKG_EXECUTABLES}
++    -DDESTINATION=${WN_PKG_DIR}
++    -DEXCLUDED_SYSTEM_PREFIXES=${CLI_WN_PKG_EXCLUDED_SYSTEM_PREFIXES}
++    -P "${CMAKE_SOURCE_DIR}/cmake/DDSCollectWNPkgFiles.cmake"
+ 	COMMAND ${CMAKE_COMMAND} -E tar czf ${WN_PKG_NAME_ARCH} "${WN_PKG_DIR}"
+-	COMMAND -chmod go+xr ${WN_PKG_NAME_ARCH}
++	COMMAND chmod go+xr ${WN_PKG_NAME_ARCH}
+ 	COMMENT "Generate WN binary package"
+ 	VERBATIM )
+ 
+@@ -450,19 +415,6 @@ add_dependencies( wn_bin_upload
+ 
+ INSTALL(FILES ${CMAKE_BINARY_DIR}/${WN_PKG_NAME_ARCH} DESTINATION bin/wn_bins OPTIONAL)
+ 
+-
+-# WORKAROUND: don't generate helper script when building an xcode project
+-# The reason: "Evaluation file to be written multiple times for different configurations or languages with different content"
+-if (NOT CMAKE_GENERATOR STREQUAL "Xcode")
+-	# WORKAROUND: create destination "lib" directory.
+-	# INSTALL SCRIPT is called before any other lib is installed, therefore we need to create a dest. dir.
+-	INSTALL(DIRECTORY DESTINATION lib)
+-	INSTALL(SCRIPT ${CMAKE_SOURCE_DIR}/cmake/modules/DDS_CollectPrerequisitesGen.cmake)
+-	if( BUILD_TESTS )
+-		INSTALL(SCRIPT ${CMAKE_SOURCE_DIR}/cmake/modules/DDS_CollectPrerequisitesGen_Tests.cmake)
+-	endif( BUILD_TESTS )
+-endif()
+-
+ #
+ # Package
+ #
+diff --git a/cmake/DDSCollectWNPkgFiles.cmake b/cmake/DDSCollectWNPkgFiles.cmake
+new file mode 100644
+index 0000000..1b31397
+--- /dev/null
++++ b/cmake/DDSCollectWNPkgFiles.cmake
+@@ -0,0 +1,24 @@
++# Copyright 2017-2020 GSI, Inc. All rights reserved.
++#
++#
++
++###################################################
++## Collect WN PKG files
++###################################################
++
++# Collect non-system dependencies
++file(GET_RUNTIME_DEPENDENCIES
++  RESOLVED_DEPENDENCIES_VAR deps
++  EXECUTABLES ${EXECUTABLES} )
++
++# Generate regex that excludes system prefixes
++list(JOIN EXCLUDED_SYSTEM_PREFIXES "|" excluded_system_prefix_regex)
++set(excluded_system_prefix_regex "^(${excluded_system_prefix_regex})/")
++
++# Copy dependencies to wn pkg staging directory
++foreach(file IN LISTS EXECUTABLES deps)
++  if(NOT file MATCHES ${excluded_system_prefix_regex})
++    message("WN PKG prerequisite='${file}'")
++    file(COPY ${file} DESTINATION ${DESTINATION})
++  endif()
++endforeach()
```

Comment:
Could you consider moving this into a `fix_wn_bin_common.patch` to reduce duplication?

Reply:
Yes, I will look into deduplicating the patches.

Reply:
So I looked over the diffs between the various patches and the part you have selected above is really the only common part which could be deduplicated. But I do not think it is meaningful to have it as a separate patch file without the calling code - it will just add overhead to generating the patch file (Right now it is straightforward, just rebase https://github.com/FairRootGroup/DDS/pull/292/commits/afe07196b3d4ccc0db580dcfe9425f46e014a40a and https://github.com/FairRootGroup/DDS/pull/292/commits/3d0aad8ff10e8ffa9cf129df0d1a5b99878cd7ed onto any given version and run e.g. `git diff -p upstream/master HEAD > ../FairSoft/packages/dds/fix_wn_bin_master.patch`) and saves a few bytes. Or, which concerns do you have if those lines are duplicated other than it costing a few bytes?

Reply:
(Btw, I made a mistake in the master patch file, it is a bit shorter now)

Reply:
It was mostly a matter of DRY.
And maybe a matter of maintainance. If we ever change the common part, we would only need to fix it once for all of them.

That said, we most likely wont have to maintain these patches forever, so I am happy with whatever works.